test/cli/test_log_runtime_archive.rb

```diff
@@ -510,6 +510,99 @@ def should_archive_dataset(dataset, archive_basename, full:)
                 end
             end
 
+            describe "#ensure_free_space" do
+                before do
+                    @archive_dir = make_tmppath
+                    @free_space_low_limit = 1
+                    @free_space_delete_until = 10
+                    @mocked_files_sizes = []
+
+                    10.times { |i| (@archive_dir / i.to_s).write(i.to_s) }
+
+                    @archiver = LogRuntimeArchive.new(
+                        @root, @archive_dir,
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                end
+
+                it "does nothing if there is enough free space" do
+                    mock_available_space(2)
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                    assert_deleted_files([])
+                end
+
+                it "removes enough files to reach the freed limit" do
+                    size_files = [6, 2, 1, 6, 7, 10, 3, 5, 8, 9]
+                    mock_files_size(size_files)
+                    mock_available_space(0.5)
+
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
```

Comment:
These are the defaults. Don't put them explicitly - but write one test where you pass non-default values and check the result.

**or** you just remove the default values passed to `#initialize`. This is probably better actually.

Reply:
as Sylvain said, you can just remove the default arguments in this call 
```suggestion
```